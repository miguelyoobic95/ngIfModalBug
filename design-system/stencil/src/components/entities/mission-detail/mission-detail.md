---
name: Mission Detail
category: Entities Components
---

### New

```mission-detail-new.html
    <yoo-mission-detail></yoo-mission-detail>
```

```mission-detail-new.js hidden
  var comp = document.querySelector('yoo-mission-detail');
  comp.mission = {
    'title': 'Store Visit Report',
    'type': 'mission',
    'priority': 4,
    'duedate': '2017-06-28T13:25:10.384Z',
    'originalUnvalidatedReason': 'Lorem ipsum dolor sit amet, consectetur adipe',
    'progress': {
      'value': 20,
      'totalPhotos': 2
    },
    'icon': 'http://res.cloudinary.com/www-yoobic-com/image/upload/v1458059035/nv6zlofgiaajrrlnxppu.jpg',
    '_geoloc': [-0.23755030000006627, 51.5314998],
    'address': '2 Bayfront Avenue B1-39A/40 & B2-38/39 The Shoppes at Marina Bay Sands, 18972, Singapore',
    'location': {
      '_id': '5836c051a501dc00176549f5',
      'clientid': 'YOOBIC_PROJECT',
      'title': 'Chandelier Building',
      'address': 'Chandelier Bldg, 8 Scrubs Ln, White City, London NW10 6RB, Royaume-Uni',
      'contactname': 'Singapore - Marina Bay Sands',
      'contactphone': '604-938-3954',
      'tags': ['test', 'CELIO_V0'],
      'info': 'Lorem ipsum dolor sit amet, consectetur adipe'
    },
    'description': {
      'text': 'Lorem ipsum dolor sit amet, consectetur adipe elit, sed do eiusmod tempor incididunt ut mare labore et dolore magna aliqua. Ut enim ad alor minim veniam, quis nostrud exercitation nerde ullamco laboris nisi ut aliquip ex ea commodo consequat.',
      'slides': [{}, {}, {}, {}, {}]
    }
  };
```
### Ongoing

```mission-detail-ongoing.html
    <yoo-mission-detail></yoo-mission-detail>
```

```mission-detail-ongoing.js hidden
  var comp = document.querySelector('yoo-mission-detail');
  comp.mission = {
    'title': 'Store Visit Report',
    'type': 'mission',
    'priority': 4,
    'duedate': '2017-06-28T13:25:10.384Z',
    'status': 'booked',
    'originalUnvalidatedReason': 'Lorem ipsum dolor sit amet, consectetur adipe',
    'progress': {
      'value': 20,
      'totalPhotos': 2
    },
    'icon': 'http://res.cloudinary.com/www-yoobic-com/image/upload/v1458059035/nv6zlofgiaajrrlnxppu.jpg',
    '_geoloc': [-0.23755030000006627, 51.5314998],
    'address': '2 Bayfront Avenue B1-39A/40 & B2-38/39 The Shoppes at Marina Bay Sands, 18972, Singapore',
    'location': {
      '_id': '5836c051a501dc00176549f5',
      'clientid': 'YOOBIC_PROJECT',
      'title': 'Chandelier Building',
      'address': 'Chandelier Bldg, 8 Scrubs Ln, White City, London NW10 6RB, Royaume-Uni',
      'contactname': 'Singapore - Marina Bay Sands',
      'contactphone': '604-938-3954',
      'tags': ['test', 'CELIO_V0'],
      'info': 'Lorem ipsum dolor sit amet, consectetur adipe'
    },
    'description': {
      'text': 'Lorem ipsum dolor sit amet, consectetur adipe elit, sed do eiusmod tempor incididunt ut mare labore et dolore magna aliqua. Ut enim ad alor minim veniam, quis nostrud exercitation nerde ullamco laboris nisi ut aliquip ex ea commodo consequat.',
      'slides': [{}, {}, {}, {}, {}]
    }
  };
```
### Pending

```mission-detail-pending.html
    <yoo-mission-detail></yoo-mission-detail>
```

```mission-detail-pending.js hidden
  var comp = document.querySelector('yoo-mission-detail');
  comp.mission = {
    'title': 'Store Visit Report',
    'type': 'mission',
    'priority': 4,
    'validatedDate': '2017-06-28T13:25:10.384Z',
    'status': 'finished',
    'originalUnvalidatedReason': 'Lorem ipsum dolor sit amet, consectetur adipe',
    'progress': {
      'value': 100,
      'totalPhotos': 2
    },
    'icon': 'http://res.cloudinary.com/www-yoobic-com/image/upload/v1458059035/nv6zlofgiaajrrlnxppu.jpg',
    '_geoloc': [-0.23755030000006627, 51.5314998],
    'address': '2 Bayfront Avenue B1-39A/40 & B2-38/39 The Shoppes at Marina Bay Sands, 18972, Singapore',
    'location': {
      '_id': '5836c051a501dc00176549f5',
      'clientid': 'YOOBIC_PROJECT',
      'title': 'Chandelier Building',
      'address': 'Chandelier Bldg, 8 Scrubs Ln, White City, London NW10 6RB, Royaume-Uni',
      'contactname': 'Singapore - Marina Bay Sands',
      'contactphone': '604-938-3954',
      'tags': ['test', 'CELIO_V0'],
      'info': 'Lorem ipsum dolor sit amet, consectetur adipe'
    },
    'description': {
      'text': 'Lorem ipsum dolor sit amet, consectetur adipe elit, sed do eiusmod tempor incididunt ut mare labore et dolore magna aliqua. Ut enim ad alor minim veniam, quis nostrud exercitation nerde ullamco laboris nisi ut aliquip ex ea commodo consequat.',
      'slides': [{}, {}, {}, {}, {}]
    }
  };
```
### Approved

```mission-detail-approved.html
    <yoo-mission-detail></yoo-mission-detail>
```

```mission-detail-approved.js hidden
  var comp = document.querySelector('yoo-mission-detail');
  comp.mission = {
    'title': 'Store Visit Report',
    'type': 'mission',
    'priority': 4,
    'validatedDate': '2017-06-28T13:25:10.384Z',
    'validated': true,
    'originalUnvalidatedReason': 'Lorem ipsum dolor sit amet, consectetur adipe',
    'progress': {
      'value': 20,
      'totalPhotos': 2
    },
    'icon': 'http://res.cloudinary.com/www-yoobic-com/image/upload/v1458059035/nv6zlofgiaajrrlnxppu.jpg',
    '_geoloc': [-0.23755030000006627, 51.5314998],
    'address': '2 Bayfront Avenue B1-39A/40 & B2-38/39 The Shoppes at Marina Bay Sands, 18972, Singapore',
    'location': {
      '_id': '5836c051a501dc00176549f5',
      'clientid': 'YOOBIC_PROJECT',
      'title': 'Chandelier Building',
      'address': 'Chandelier Bldg, 8 Scrubs Ln, White City, London NW10 6RB, Royaume-Uni',
      'contactname': 'Singapore - Marina Bay Sands',
      'contactphone': '604-938-3954',
      'tags': ['test', 'CELIO_V0'],
      'info': 'Lorem ipsum dolor sit amet, consectetur adipe'
    },
    'description': {
      'text': 'Lorem ipsum dolor sit amet, consectetur adipe elit, sed do eiusmod tempor incididunt ut mare labore et dolore magna aliqua. Ut enim ad alor minim veniam, quis nostrud exercitation nerde ullamco laboris nisi ut aliquip ex ea commodo consequat.',
      'slides': [{}, {}, {}, {}, {}]
    }
  };
```

### Rejected

```mission-detail-rejected.html
    <yoo-mission-detail></yoo-mission-detail>
```

```mission-detail-rejected.js hidden
  var comp = document.querySelector('yoo-mission-detail');
  comp.mission = {
    'title': 'Store Visit Report',
    'type': 'mission',
    'priority': 4,
    'validatedDate': '2017-06-28T13:25:10.384Z',
    'validated': false,
    'originalUnvalidatedReason': 'Lorem ipsum dolor sit amet, consectetur adipe',
    'progress': {
      'value': 20,
      'totalPhotos': 2
    },
    'icon': 'http://res.cloudinary.com/www-yoobic-com/image/upload/v1458059035/nv6zlofgiaajrrlnxppu.jpg',
    '_geoloc': [-0.23755030000006627, 51.5314998],
    'address': '2 Bayfront Avenue B1-39A/40 & B2-38/39 The Shoppes at Marina Bay Sands, 18972, Singapore',
    'location': {
      '_id': '5836c051a501dc00176549f5',
      'clientid': 'YOOBIC_PROJECT',
      'title': 'Chandelier Building',
      'address': 'Chandelier Bldg, 8 Scrubs Ln, White City, London NW10 6RB, Royaume-Uni',
      'contactname': 'Singapore - Marina Bay Sands',
      'contactphone': '604-938-3954',
      'tags': ['test', 'CELIO_V0'],
      'info': 'Lorem ipsum dolor sit amet, consectetur adipe'
    },
    'description': {
      'text': 'Lorem ipsum dolor sit amet, consectetur adipe elit, sed do eiusmod tempor incididunt ut mare labore et dolore magna aliqua. Ut enim ad alor minim veniam, quis nostrud exercitation nerde ullamco laboris nisi ut aliquip ex ea commodo consequat.',
      'slides': [{}, {}, {}, {}, {}]
    }
  };
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`mission`|IMission|*required|contains the informations related to the mission|
|`slidesNumber`|string|undefined|the number of steps|
|`photosNumber`|string|undefined|the number of photos|
|`questionsNumber`|string|undefined|the number of question|