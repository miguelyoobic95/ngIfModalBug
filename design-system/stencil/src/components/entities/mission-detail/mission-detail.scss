:host {
    height: 100%;
    font-size: 0.9375rem;
    font-weight: 400;
    .mission-detail {
        height: 100%;
        .mission-content {
            padding: 1rem;
            .mission-title {
                font-weight: 600;
                font-size: 1.6875rem;
                line-height: 2.0625rem;
                margin-top: 0.3125rem;
                margin-bottom: 0.3125rem;
            }
            .mission-date {
                line-height: 1.125rem;
                font-weight: 300;
                color: $text-color;
            }
            .mission-progress {
                margin-top: 0.3125rem;
            }
            .mission-menu {
                list-style: none;
                padding: 0;
                margin-top: 1.875rem;
                .mission-menu-item {
                    display: flex;
                    flex-direction: row;
                    .mission-menu-left {
                        display: flex;
                        flex-direction: column;
                        align-items: center;
                        .mission-menu-icon {
                            height: 1.875rem;
                            width: 1.875rem;
                            i {
                                font-size: 1.0625rem;
                                display: flex;
                                justify-content: center;
                                background-color: cssvar(light);
                                border: solid 0.03125rem $border-stable;
                                border-radius: 50%;
                                padding: 0.34375rem;
                            }
                        }
                        .border {
                            width: 1px;
                            height: 100%;
                            border: solid 0.03125rem $border-stable;
                        }
                    }
                    .mission-menu-right {
                        margin-left: 0.625rem;
                        .mission-menu-title {
                            font-weight: 600;
                            line-height: 1.125rem;
                            margin-top: 0.375rem;
                            margin-bottom: 0.375rem;
                        }
                        .mission-menu-content {
                            margin-top: 0.25rem;
                            margin-bottom: 1.25rem;
                            span {
                                margin-right: 1.25rem;
                            }
                            .mission-menu-content-contact {
                                margin-bottom: 0.625rem;
                            }
                        }
                    }
                }
                .mission-menu-item:last-child {
                    .mission-menu-left {
                        .border {
                            display: none;
                        }
                    }
                }
            }
        }
        .mission-footer {
            display: flex;
            flex-direction: row;
            justify-content: space-around;
            padding-top: 1.25rem;
            padding-bottom: 1.4rem;
            box-shadow: $input-bar-shadow;
        }
    }
}