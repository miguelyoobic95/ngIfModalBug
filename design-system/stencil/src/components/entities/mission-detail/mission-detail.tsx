import { Component, Element, Prop, Method, State, EventEmitter, Event } from '@stencil/core';
import { YooSlimScrollComponent } from '../../core/slim-scroll/slim-scroll';
import { IMission, IMissionDescription, ILocation, IUser, ITranslateService } from '@shared/interfaces';
import { resizeWindow } from '../../../utils/helpers';
import { pipes } from '../../../utils/pipes';

@Component({
    tag: 'yoo-mission-detail',
    styleUrl: 'mission-detail.scss',
    scoped: true
})
export class YooMissionDetailComponent {

    @Prop() mission: IMission;
    @Prop() slidesNumber: number;
    @Prop() photosNumber: number;
    @Prop() questionsNumber: number;

     @Element() host: HTMLStencilElement;
    @Event() book: EventEmitter<string>;

    @State() scrollHeight: string;

    private badgeText: string;
    private badgeClass: string;
    private progressClass: string;
    private translate: ITranslateService = (window as any).translateService;

    private newStatus: string = 'NEW';
    private bookedStatus: string = 'BOOKED';
    private pendingStatus: string = 'PENDING';

    @Method()
    resize() {
        let slim: YooSlimScrollComponent = this.host.querySelector('yoo-slim-scroll') as any;
        if (slim) {
            this.scrollHeight = this.getSizeContainer().height;
        }
    }

    componentWillLoad() {
        this.badgeText = null;
        this.badgeClass = '';
        this.progressClass = undefined;

        if (this.mission != null) {
            switch (this.mission['validated']) {
                case true: {
                    this.badgeText = this.translate.get('VALIDATED');
                    this.badgeClass = 'gradient-success';
                    this.progressClass = undefined;
                    break;
                }
                case false: {
                    this.badgeText = this.translate.get('REJECTED');
                    this.badgeClass = 'danger';
                    this.progressClass = undefined;
                    break;
                }
                default: {
                    switch (this.mission['status']) {
                        case undefined: {
                            this.badgeText = this.newStatus;
                            this.badgeClass = 'gradient-success';
                            this.progressClass = 'dark';
                            break;
                        }
                        case 'booked': {
                            this.badgeText = this.bookedStatus;
                            this.badgeClass = 'info';
                            this.progressClass = 'info';
                            break;
                        }
                        case 'finished': {
                            this.badgeText = this.pendingStatus;
                            this.badgeClass = 'warning';
                            this.progressClass = 'warning';
                            break;
                        }
                        default: {
                            break;
                        }
                    }
                    break;
                }
            }
        }
    }

    componentDidLoad() {
        resizeWindow(() => this.resize());
        this.resize();
    }

    getSizeContainer() {
        let maxHeight = window.innerHeight;
        let body = this.host.parentElement.parentElement.querySelector('.modal-body');
        if (body) {
            maxHeight = Math.min(maxHeight, body.clientHeight);
        } else {
            maxHeight = this.decreaseMaxHeight(maxHeight, 'ion-header', document);
            maxHeight = this.decreaseMaxHeight(maxHeight, 'ion-footer', document);
        }
        maxHeight = this.decreaseMaxHeight(maxHeight, '.mission-footer', this.host);
        return { height: maxHeight + 'px' };
    }

    decreaseMaxHeight(maxHeight: number, name: string, html) {
        let comp: HTMLElement = html.querySelector(name);
        if (comp) {
            maxHeight -= comp.clientHeight;
        }
        return maxHeight;
    }

    onBook(action: string) {
        this.book.emit(action);
    }

    renderButtonFooter(): JSX.Element {
        return (
            <div class="mission-footer">
                {(this.badgeText === this.newStatus) ? <yoo-button onClick={() => this.onBook('start')} text={this.translate.get('START')} class="large gradient-success"></yoo-button> : null}
                {(this.badgeText === this.bookedStatus) ? <yoo-button onClick={() => this.onBook('unbook')} text={this.translate.get('UNBOOK')} class="mission-button-leave"></yoo-button> : null}
                {(this.badgeText === this.bookedStatus) ? <yoo-button onClick={() => this.onBook('start')} text={this.translate.get('CONTINUE')} class="gradient-success"></yoo-button> : null}
            </div>
        );
    }

    render(): JSX.Element {
        let description = {} as IMissionDescription, location = {} as ILocation, creator = {} as IUser;
        if (this.mission) {
            description = this.mission.description as IMissionDescription;
            location = this.mission.location as ILocation;
            creator = this.mission.creator as IUser;
        }
        return (this.mission ?
            <div class="mission-detail">
                <yoo-slim-scroll height={this.scrollHeight}>
                    <div class="mission-content">
                        {(this.badgeText ? <yoo-badge text={this.translate.get(this.badgeText || '')} class={'round small ' + this.badgeClass}></yoo-badge> : null)}
                        {(this.mission.title ? <div class="mission-title">{this.translate.polyglot(this.mission.title)}</div> : null)}
                        {(this.mission.duedate ?
                            <div class="mission-date">{this.translate.get('DUEDATE')}: {pipes.dateFormat.transform(this.mission.duedate, 'L LT')}</div>
                            :
                            (this.mission.validatedDate ? <div class="mission-date">{this.translate.get('VALIDATEDDATE')}: {this.mission.validatedDate}</div> : null)
                        )}
                        {(this.progressClass ? <div class="mission-progress"><yoo-progress-bar progress={(this.mission.progress ? this.mission.progress.value : 0)} class={'xsmall ' + this.progressClass}></yoo-progress-bar></div> : null)}
                        <ul class="mission-menu">
                            {(this.mission.priority ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-priority"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('PRIORITY')}</div>
                                        <div class="mission-menu-content">P{this.mission.priority}</div>
                                    </div>
                                </li>
                                : null)}
                            {(description.text ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-description"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('MISSIONDESCRIPTION')}</div>
                                        <div class="mission-menu-content">{description.text}</div>
                                    </div>
                                </li>
                                : null)}
                            {((this.slidesNumber || this.photosNumber || this.questionsNumber) ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-attachment"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('QUESTIONS')}</div>
                                        <div class="mission-menu-content">
                                            {(this.slidesNumber ?
                                                <span>
                                                    <i class="yo-pages"></i> {this.slidesNumber} {this.translate.get('PAGES')}
                                                </span> : null)}
                                            {(this.photosNumber ?
                                                <span>
                                                    <i class="yo-gallery"></i> {this.photosNumber} {this.translate.get('PHOTOS')}
                                                </span> : null)}
                                            {(this.questionsNumber ?
                                                <span>
                                                    <i class="yo-questions"></i> {this.questionsNumber} {this.translate.get('QUESTIONS')}
                                                </span> : null)}
                                        </div>
                                    </div>
                                </li>
                                : null)}
                            {((location.contactname || location.contactphone || this.mission.address) ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-contact"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('CONTACT')}</div>
                                        <div class="mission-menu-content">
                                            {(location ? <div class="mission-menu-content-contact">{location.contactname}</div> : null)}
                                            {(this.mission.address ? <div class="mission-menu-content-contact">{this.mission.address}</div> : null)}
                                            {(location ? <div class="mission-menu-content-contact">{location.contactphone}</div> : null)}
                                        </div>
                                    </div>
                                </li>
                                : null)}
                            {(this.mission.comments ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-note"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('NOTES')}</div>
                                        <div class="mission-menu-content">{this.mission.comments}</div>
                                    </div>
                                </li>
                                : null)}
                            {(location.info ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-info"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('INFO')}</div>
                                        <div class="mission-menu-content">{location.info}</div>
                                    </div>
                                </li>
                                : null)}
                            {(this.mission.originalUnvalidatedReason ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-comment"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('COMMENTS')}</div>
                                        <div class="mission-menu-content">{this.mission.originalUnvalidatedReason}</div>
                                    </div>
                                </li>
                                : null)}
                            {(this.mission.price ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-plus"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('PRICE')}</div>
                                        <div class="mission-menu-content">{this.mission.price}</div>
                                    </div>
                                </li>
                                : null)}
                            {(creator && (creator.email || creator.firstName || creator.lastName) ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-plus"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('REQUESTOR')}</div>
                                        {((creator.firstName || creator.lastName) ? <div class="mission-menu-content">{creator.firstName} {creator.lastName}</div> : null)}
                                        {(creator.email ? <div class="mission-menu-content">{creator.email}</div> : null)}
                                    </div>
                                </li>
                                : null)}
                            {((this.mission.duration) ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-plus"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('MISSIONDURATION')}</div>
                                        <div class="mission-menu-content">{this.mission.duration} min</div>
                                    </div>
                                </li>
                                : null)}
                            {((this.mission._ect) ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-plus"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('CREATIONDATE')}</div>
                                        <div class="mission-menu-content">{pipes.dateFormat.transform(this.mission._ect, 'L LT')}</div>
                                    </div>
                                </li>
                                : null)}
                            {((this.mission.validFrom || this.mission.validUntil) ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-plus"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('VISIBILITY')}</div>
                                        <div class="mission-menu-content">{pipes.dateFormat.transform(this.mission.validFrom, 'L LT')} - {pipes.dateFormat.transform(this.mission.validUntil, 'L LT')}</div>
                                    </div>
                                </li>
                                : null)}
                            {(this.mission.bookedUntil ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-plus"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('CALENDAR')}</div>
                                        <div class="mission-menu-content">{pipes.dateFormat.transform(this.mission.bookedUntil, 'L LT')}</div>
                                    </div>
                                </li>
                                : null)}
                            {(this.mission.serviceData ?
                                <li class="mission-menu-item">
                                    <div class="mission-menu-left">
                                        <span class="mission-menu-icon"><i class="yo-plus"></i></span>
                                        <div class="border"></div>
                                    </div>
                                    <div class="mission-menu-right">
                                        <div class="mission-menu-title">{this.translate.get('DETAILS')}</div>
                                        <div class="mission-menu-content">{this.mission.serviceData}</div>
                                    </div>
                                </li>
                                : null)}
                        </ul>
                    </div>
                </yoo-slim-scroll>
                {((this.badgeText === this.newStatus || this.badgeText === this.bookedStatus) ? this.renderButtonFooter() : null)}
            </div>
            : null
        );
    }
}
