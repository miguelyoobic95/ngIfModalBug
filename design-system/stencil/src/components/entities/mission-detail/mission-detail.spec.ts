import { TestWindow } from '@stencil/core/testing';
import { YooMissionDetailComponent } from './mission-detail';
// import { YooBadgeComponent } from '../../core/badge/badge';
// import { YooProgressBarComponent } from '../../core/progress-bar/progress-bar';

describe('MissionDetailComponent', () => {
    let data = {
    'title': 'Store Visit Report',
    'type': 'mission',
    'priority': 4,
    'duedate': '2017-06-28T13:25:10.384Z',
    'status': 'booked',
    'originalUnvalidatedReason': 'Lorem ipsum dolor sit amet, consectetur adipe',
    'progress': {
      'value': 20,
      'totalPhotos': 2
    },
    'icon': 'http://res.cloudinary.com/www-yoobic-com/image/upload/v1458059035/nv6zlofgiaajrrlnxppu.jpg',
    '_geoloc': [-0.23755030000006627, 51.5314998],
    'address': '2 Bayfront Avenue B1-39A/40 & B2-38/39 The Shoppes at Marina Bay Sands, 18972, Singapore',
    'location': {
      '_id': '5836c051a501dc00176549f5',
      'clientid': 'YOOBIC_PROJECT',
      'title': 'Chandelier Building',
      'address': 'Chandelier Bldg, 8 Scrubs Ln, White City, London NW10 6RB, Royaume-Uni',
      'contactname': 'Singapore - Marina Bay Sands',
      'contactphone': '604-938-3954',
      'tags': ['test', 'CELIO_V0'],
      'info': 'Lorem ipsum dolor sit amet, consectetur adipe'
    },
    'description': {
      'text': 'Lorem ipsum dolor sit amet, consectetur adipe elit, sed do eiusmod tempor incididunt ut mare labore et dolore magna aliqua. Ut enim ad alor minim veniam, quis nostrud exercitation nerde ullamco laboris nisi ut aliquip ex ea commodo consequat.',
      'slides': [{}, {}, {}, {}, {}]
    }
  };

    it('should build', () => {
        expect(new YooMissionDetailComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let element, window;
        beforeEach(async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooMissionDetailComponent/**, YooProgressBarComponent, YooBadgeComponent**/],
                html: '<yoo-mission-detail></yoo-mission-detail>'
            });
        });

        it('Should render', async () => {
            expect(element).toMatchSnapshot();
        });
        it('Should render with mission', async () => {
            element.mission = data;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
    });
    describe('Events', () => {
        let emitEvent, window;
        beforeEach(async () => {
            window = new TestWindow();
            emitEvent = jest.fn();
        });
    });
});