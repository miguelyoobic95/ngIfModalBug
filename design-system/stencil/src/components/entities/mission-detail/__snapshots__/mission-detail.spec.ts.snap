// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionDetailComponent Rendering Should render 1`] = `
<yoo-mission-detail
  class="hydrated"
  data-yoo-mission-detail-host=""
>
  
  
</yoo-mission-detail>
`;

exports[`MissionDetailComponent Rendering Should render with mission 1`] = `
<yoo-mission-detail
  class="hydrated"
  data-yoo-mission-detail-host=""
>
  
  
</yoo-mission-detail>
`;
